sympy/stats/stochastic_process_types.py

```diff
@@ -615,6 +654,20 @@ class DiscreteMarkovChain(DiscreteTimeStochasticProcess, MarkovProcess):
     >>> P(Eq(Y[3], 2), Eq(Y[1], 1)).round(2)
     0.36
 
+    Probabilities will be calculated based on indexes rather
+    than state names. For example, with the Sunny-Cloudy-Rainy
+    model:
+
+    >>> Y = DiscreteMarkovChain("Y", ['Sunny', 'Cloudy', 'Rainy'], T)
+    >>> P(Eq(Y[3], 2), Eq(Y[1], 1)).round(2)
+    0.36
+
+    This gives the same answer as the ``[0, 1, 2]`` state space.
+    Currently, there is no support for state names within probability
+    and expectation statements. It might be supported in the future
+    in the form of a keyword argument ``names=True``.
+    (how would P(X > 'Sunny') work?)
```

Comment:
Does `P(Gt(X, 'Sunny'))` work?

Reply:
No. Probability statements are done using the state indexes (eg 0,1,..., n). Explanation is in the docstring. I would have to really dig into the code if I wanted to be able to do it using state names. I'm not even sure if `Eq` or other relationals even work with strings.